### PR TITLE
Avoid using RARRAY_PTR

### PR DESCRIPTION
**Pull request by casperisfine**
It cause the array to be marked as unprotected, when this happen enough times it triggers a major GC run.

### CODE REVIEW

File: ext/rlibmemcached/rlibmemcached.i [hunk 1]
```diff
@@ -47,7 +47,9 @@
   $2 = (size_t *) malloc(($3+1)*sizeof(size_t));
   $1 = (char **) malloc(($3+1)*sizeof(char *));
   for(i = 0; i < $3; i ++) {
-    str = rb_string_value(&RARRAY_PTR($input)[i]);
+    str = rb_ary_entry(argv[1], i);
```
Comment (peterzhu2118) [inline ext/rlibmemcached/rlibmemcached.i]: What's the difference between `argv[1]` and `$input`?

File: ext/rlibmemcached/rlibmemcached.i [hunk 2]
```diff
@@ -47,7 +47,9 @@
...
+    str = rb_string_value(&str);
```
Comment (peterzhu2118) [inline ext/rlibmemcached/rlibmemcached.i]: `rb_string_value` already writes to the pointer:

```suggestion
    rb_string_value(&str);
```

File: ext/rlibmemcached/rlibmemcached.i (hunk 1 above)
Comment (casperisfine) [reply ext/rlibmemcached/rlibmemcached.i]: `$input` was the correct thing, this is some kind of binding generation language, it's mostly dead code at this point though, but might as well keep in it hcekc.

Good catch.

File: ext/rlibmemcached/rlibmemcached.i (hunk 2 above)
Comment (casperisfine) [reply ext/rlibmemcached/rlibmemcached.i]: Fixed.

File: ext/rlibmemcached/rlibmemcached_wrap.c [hunk 3]
```diff
@@ -9104,7 +9105,9 @@ _wrap_memcached_mget_by_key(int argc, VALUE *argv, VALUE self) {
     arg5 = (size_t *) malloc((arg6+1)*sizeof(size_t));
     arg4 = (char **) malloc((arg6+1)*sizeof(char *));
     for(i = 0; i < arg6; i ++) {
-      str = rb_string_value(&RARRAY_PTR(argv[3])[i]);
+      str = rb_ary_entry(argv[1], i);
```
Comment (peterzhu2118) [inline ext/rlibmemcached/rlibmemcached_wrap.c:9108]: This was `argv[3]` before and it's now `argv[1]`?

Comment (casperisfine) [reply ext/rlibmemcached/rlibmemcached_wrap.c:9108]: Woops. I'm surprised the test suite pass, but not sure how much we can trust it. Thanks for the catch.

Comment (casperisfine) [reply ext/rlibmemcached/rlibmemcached_wrap.c:9108]: Fixed in 28b5840461f459986953660a3079f3547b1ce1d2

**Pull request merged**